pkg/probe/http/http_test.go

```diff
@@ -87,13 +90,156 @@ func TestHTTPProbeProxy(t *testing.T) {
+		health            probe.Result
+		expectHTTPCodes   []int
+		expectHTTPContent string
+		accBody           string
+		notBody           string
+	}{
+		// The probe will be filled in below.  This is primarily testing that an HTTP GET happens.
+		{
+			handler:           handleReq(http.StatusOK, "helloworld"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "helloworld",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusOK, "ok"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404},
+			expectHTTPContent: "",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{500, 404},
+			expectHTTPContent: "",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{200},
+			expectHTTPContent: "",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{400, 200},
+			expectHTTPContent: "",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404, 500},
+			expectHTTPContent: "not*",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404, 500},
+			expectHTTPContent: "found*",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{200},
+			expectHTTPContent: "not *",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404, 500},
+			expectHTTPContent: "not *",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "not *",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "found *",
+			health:            probe.Failure,
+		},
+	}
+
+	for i, test := range testCases {
+		t.Run(fmt.Sprintf("case-%2d", i), func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				test.handler(w, r)
+			}))
+			defer server.Close()
+			u, err := url.Parse(server.URL)
+			if err != nil {
+				t.Errorf("case %d: unexpected error: %v", i, err)
```

Comment:
As we use `t.Run()` that has information about current test, I'm not sure that we should include a test case name also in the error message.

Also, consider using ``t.Fatalf()` where it possible.

Reply:
done